src/metrics/representation.py

```diff
@@ -0,0 +1,381 @@
+            targets: The vectors targets, size (batch_size).
+            queries_idxs: Integer tensor of query number in order, if it equal -1 - vector is not query.
+            scores: The scores tensor, see representation dataset for more information, 
+                size (batch_size, all_query_count).
+
+        Raises:
+            ValueError: If dataset is classification and targets is None, or if dataset is representation and one of
+                scores or queries_idxs is None.
+        """
+        vectors = vectors.detach().cpu()
+        self.vectors.append(vectors)
+        if self.dataset == 0:
+            if targets is None:
+                raise ValueError("In classification dataset target must be a not None value.")
+            targets = targets.detach().cpu()
+            self.targets.append(targets)
+        else:
+            if queries_idxs is None:
+                raise ValueError("In representation dataset queries_idxs must be a not None value.")
+            if scores is None:
+                raise ValueError("In representation dataset scores must be a not None")
+            
+            queries_idxs = queries_idxs.detach().cpu()
+            self.queries_idxs.append(queries_idxs)
+            
+            scores = scores.detach().cpu()
+            self.scores.append(scores)
+
+    def compute(self) -> Generator[List[np.ndarray], List[np.ndarray]]:
+        """Build generator with relevant, closest arrays.
+        
+        Firstly it gather all storage.
+        Prepare data, separate queries vectors and data base vectors.
+        Then build faiss index.
+        Create generator of relevant, closest arrays.
+
+        Returns:
+            Generator wich return relevant and closest examples.
+        """
+        vectors = torch.cat(self.vectors).numpy()
+        
+        if self.normalize_input:
+            vectors = normalize(vectors)
+
+        if self.dataset == 0:
+            # if classification dataset
+            targets = torch.cat(self.targets).numpy()
+            # prepare data
+            q_vecs, db_vecs, relevants, scores, \
+                db_idxs, q_order_idxs = self.prepare_classification_data(vectors, targets)
+        else:
+            # if representation dataset
+            scores = torch.cat(self.scores).numpy()
+            # convert numpy array with type long to bool
+            queries_idxs = torch.cat(self.queries_idxs).numpy()
+            # prepare data
+            q_vecs, db_vecs, relevants, scores, \
+                db_idxs, q_order_idxs = self.prepare_representation_data(vectors, queries_idxs, scores)
+
+        q_vecs = q_vecs.astype(np.float32)
+        db_vecs = db_vecs.astype(np.float32)
+
+        index = self.build_index(db_vecs)
+
+        # if search batch size is None, search queries vectors by one request
+        search_batch_size = len(q_vecs) if self.search_batch_size is None else self.search_batch_size
```

Comment:
It might be overload the memory, if request size is large and vectors sizes is also large. What do you suggest? Set as default search_batch_size = num (8)?

Reply:
Yes, something like this. As far as I know, `compute` is called in a single process. So, you can safely set `batch_size` equal to number of CPUs